app/models/action.rb

```diff
@@ -8,7 +8,18 @@ class Action < ApplicationRecord
   DENY_OPERATION    = 'deny'.freeze
   OPERATIONS = [NOTIFY_OPERATION, SKIP_OPERATION, MEMO_OPERATION, APPROVE_OPERATION, DENY_OPERATION].freeze
 
+  DATE_ATTRIBUTES     = %w[created_at updated_at].freeze
+  NON_DATE_ATTRIBUTES = %w[processed_by operation comments stage_id].freeze
+
   validates :operation,    :inclusion => { :in => OPERATIONS }
   validates :processed_by, :presence  => true
   belongs_to :stage, :inverse_of => :actions
+
+  def as_json(_options = {})
+    attributes.slice(*NON_DATE_ATTRIBUTES).tap do |hash|
+      DATE_ATTRIBUTES.each do |attr|
+        hash[attr] = send(attr.to_sym).iso8601 if send(attr.to_sym)
+      end
+    end.merge(:id => id.to_s)
```

Comment:
Constants are the wrong way to go here.

You can check for the a date-time field a few ways:

1) Check the type of column:
```ruby
pry(#<Service>):1> column_for_attribute('created_at').type
=> :datetime
```

2) Checking the type of value:
```ruby
pry(#<Service>):1> attribute('created_at').kind_of?(Time)
=> true
```

@syncrou @mkanoor Do we have `as_json` logic anywhere else that we should be sharing here?  Seems we are just converting date fields, making `id` a string and dropping `tenant_id`.

Also, should `as_json` be using symbols?  It feels like the whole thing could be a small method build on `attributes.except('tenant_id')`.

And the duplication makes me think mixin.

Reply:
@gmcculloug 
We have something in topology that needs to be exposed to all services where we read the openapi.json file and look at the response object and convert it to the valid strings. The  list would be defined in the openapi.json

https://github.com/ManageIQ/topological_inventory-api/blob/3300b0262aa686e0e3c29c77e54235b5c30425b5/lib/open_api/serializer.rb#L3

I think @syncrou had started working on it at some point I am hoping it will get integrated into  manageiq-api-common

Reply:
@syncrou Any thoughts here?  Should we copy the Topology pattern here until we can get it into common code?

Reply:
@gmcculloug I just made changes based on your previous comments.